components/pandadoc/actions/send-document/send-document.mjs

```diff
@@ -61,6 +61,24 @@ export default defineComponent({
   async run({ $ }) {
     const documentId = this.documentId;
 
+    let documentStatus = "";
+
+    while (documentStatus !== "document.draft") {
+      try {
+        const response = await this.app.getDocument({
+          id: documentId,
+        });
+        docStatus = response.status;
+        if (docStatus == "document.draft") break;
+        console.log(
+          `Document status is '${documentStatus}' and not 'document.draft'. Waiting 1s and trying again...`
+        );
+        await new Promise((resolve) => setTimeout(resolve, 1000)); // Wait for 1 second
```

Comment:
Is it necessary to have this as 1 second and unlimited requests?

I'd wait for 5 seconds, and if it makes sense for the business rule here, establish a max number of attempts before dropping the loop.

Reply:
@GTFalcao I guess so, when I was testing it always worked after a second or two.

Even at unlimited requests it would just hit a timeout.